### DISCUSSION TITLE
How can I use `HFS.dialogLib`?

### DISCUSSION THREAD

**Discussion by PyroSoar** [rejetto/hfs#519] [Q&A] [answered]
I am trying to write a plugin, but i can't find documents about `HFS.dialogLib`(like including html content, title, defining callbacks, etc). Can anyone help? :)

**Comment by rejetto** [comment]
that part is not documented, but exposed because can be handy.
basically, dialogLib exposes everything that you see in this file
https://github.com/rejetto/hfs/blob/main/frontend/src/dialog.ts

what kind of dialog are you trying to get?
newDialog is the generic one, but you have several specialized dialogs

**Comment by PyroSoar** [comment]
Thans for your reply!
I failed to use the simplest alert containing html tags, say `HFS.dialogLib.alertDialog("<b>test</b>")`. How to contain html tags in messages?

**Comment by rejetto** [comment, marked as answer]
html as a string is not supported.
HFS is using React. In React you build every dom node. You must use function HFS.h to do that:
HFS('b', {}, 'test')

**Reply by PyroSoar** [reply]
Thank you for your reply! This workaround can achieve basic functions, but a more convenient method is still expected.

**Reply by rejetto** [reply]
nope, this method can make anything. You can make anything with React. That's how I made the whole software. 😄
WHat are you missing?

**Reply by PyroSoar** [reply]
Oh, I see your point! Sorry about the misunderstanding earlier.

**Comment by rejetto** [comment]
i will consider something easier for the future
